dsl/src/main/scala/org/http4s/dsl/Path.scala

```diff
@@ -31,11 +32,8 @@ object Path {
       Path("/" + str)
     else {
       val slash = str.lastIndexOf('/')
-      val prefix = Path(str.substring(0, slash))
-      if (slash == str.length - 1)
-        prefix
-      else
-        prefix / str.substring(slash + 1)
+      val prefix = Path(UrlCodingUtils.urlDecode(str.substring(0, slash)))
```

Comment:
I think I see a bug that will cause problems with ```%2F``` in the path elements. I see you have a test for this below in the form of ```"/foo%20bar/1%2F2"``` that passes, but what about ```"/foo%2Fbar/blah"```? I suspect that will end up structured as ```Root / "foo" / "bar" / "blah"``` as opposed to the intended ```Root / "foo/bar" / "blah"```. If so, the culprit is urldecoding the argument here.
If not, my apologies :smile: